pkg/daemon/ceph/osd/device.go

```diff
@@ -501,7 +501,10 @@ func WriteConfigFile(context *clusterd.Context, cluster *cephconfig.ClusterInfo,
 }
 
 func writeConfigFile(cfg *osdConfig, context *clusterd.Context, cluster *cephconfig.ClusterInfo, location string) error {
-	cephConfig := cephconfig.CreateDefaultCephConfig(context, cluster, cfg.rootPath)
+	cephConfig, err := cephconfig.CreateDefaultCephConfig(context, cluster, cfg.rootPath)
+	if err != nil {
+		return fmt.Errorf("failed to create default ceph config. %+v", err)
```

Comment:
nit: i think the pattern of returning `fmt.Errorf` where the string is formatted with a lower-level `err` makes debugging harder because the error strings get concatenated together. It's definitely dependent on the situation, but I might be tempted here to either (1) return `err` without additional formatting or (2) `logger.Error("failed to create..."); return err;`

Reply:
While I would agree it makes debugging harder, I think it is best to keep with the existing format, both for the sake of consistency and also that it makes it clear that the failure cascades and that the failures aren't two unrelated events.

Reply:
@noahdesu I've found that encapsulation a bit annoying too but like @BlaineEXE said this is how things are structured.